Standards/scs-0117-v1-rook-decision.md

```diff
@@ -0,0 +1,76 @@
+
+## Context
+
+The current reference implementation relies on `ceph-ansible`, [which is now deprecated](https://github.com/ceph/ceph-ansible/commit/a9d1ec844d24fcc3ddea7c030eff4cd6c414d23d). As a result, this decision record evaluates two alternatives: Cephadm and Rook.
+
+Both tools are designed to roll out and configure Ceph clusters, providing the capability to manage clusters throughout their lifecycle. This includes functionalities such as adding or removing OSDs, upgrading Ceph services, and managing CRUSH maps, as outlined in the [Feature-Decision-Table](#feature-decision-table).
+
+This decision record considers both the current and future needs of the reference implementation. The decision is guided by a comprehensive comparison of each tool's capabilities and limitations as well as the SCS communities needs and futures objectives.
+
+### Comparison of Features
+
+The tool selected in this decision MUST ensure:
+
+* ease of migration
+* future-proofness
+* feature-completeness and feature-maturity
+* effective management of Ceph clusters
+
+#### Feature Decision Table
+
+A comparative analysis of Cephadm and Rook highlights the following:
+
+| Feature | Supported in Cephadm | Supported in Rook |
+| ------- | -------------------- | ----------------- |
+| Migrate from other setups | ☑ Adoption of clusters, that where built with ceph-ansible [is officially supported](https://docs.ceph.com/en/quincy/cephadm/adoption/).| ☐ Migration from other setups is not offically supported. See this [issue](https://github.com/rook/rook/discussions/12045). Consequently, SCS develops a migration tool, named [rookify](https://github.com/SovereignCloudStack/rookify). Alternatively, Rook allows to use [Ceph as an external cluster](https://rook.io/docs/rook/latest-release/CRDs/Cluster/external-cluster/external-cluster/). |
+| Connect RGW with OpenStack Keystone | ☑ | ☑ Experimental |
+| Deploy specific Ceph versions | ☑  | ☑ |
+| Upgrade to specific Ceph versions | ☑ Streamlined upgrade process. | ☑ Rook, CSI and Ceph upgrades have to be aligned, there is a [guide](https://rook.io/docs/rook/latest-release/Upgrade/health-verification/) available for each Rook version. |
+| Deploy Ceph Monitors | ☑ | ☑ |
+| Deploy Ceph Managers | ☑ | ☑ |
+| Deploy Ceph OSDs | ☑ | ☑ |
+| Deploy Ceph Object Gateway (RGW) | ☑ | ☑ |
+| Removal of nodes | ☑ | ☑ |
+| Purging of complete cluster | ☑ | ☑ |
+
+☐ not supported (yet)
+☑ supported
+☑☑ better option
+☒ not supported on purpose
+
+#### Evaluation in the Light of SCS Community Plans and Preferences
+
+**Environment**: Cephadm is better suited for traditional or standalone environments. Conversely, Rook is tailored for Kubernetes. That being said, it's important to note that the current state of resource deployment and management on Kubernetes within the reference implementation is still in its early stages. This would make Rook one of the first components to utilise Kubernetes in OSISM.
```

Comment:
```suggestion
**Environment**: Cephadm is better suited for traditional or standalone environments. Conversely, Rook is tailored for Kubernetes. That being said, it's important to note that the current state of resource deployment and management on Kubernetes within the IaaS reference implementation is still in its early stages. This would make Rook one of the first components to utilise Kubernetes in OSISM.
```

Furthermore KeyCloak is already being deployed to the kubernetes, so I'm not sure wether the statement of the Rook being the first holds. Nevertheless, it is early stages, so that of course is correct.

Reply:
I'd say that `one of the first components` states that Rook isn't the very fist component on Kubernetes.